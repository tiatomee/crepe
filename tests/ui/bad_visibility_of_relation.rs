// In this test the visibility modifiers of relation structs and their fields
// are checked to be translated into the final Rust code properly.

mod datalog {
    use crepe::crepe;

    crepe! {
        @input
        struct Test(u32);

        @input
        pub struct MoreTest(bool);

        @input
        pub struct FullyPublic(pub i8);
    }
}

fn main() {
    let _ = datalog::FullyPublic(21);
    let _ = datalog::Test(1);
    let _ = datalog::MoreTest(false);
}
