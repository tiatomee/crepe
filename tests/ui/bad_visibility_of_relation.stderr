error[E0603]: tuple struct constructor `Test` is private
  --> $DIR/bad_visibility_of_relation.rs:21:22
   |
9  |         struct Test(u32);
   |                     --- a constructor is private if any of the fields is private
...
21 |     let _ = datalog::Test(1);
   |                      ^^^^ private tuple struct constructor
   |
note: the tuple struct constructor `Test` is defined here
  --> $DIR/bad_visibility_of_relation.rs:9:9
   |
9  |         struct Test(u32);
   |         ^^^^^^^^^^^^^^^^^

error[E0603]: tuple struct constructor `MoreTest` is private
  --> $DIR/bad_visibility_of_relation.rs:22:22
   |
12 |         pub struct MoreTest(bool);
   |                             ---- a constructor is private if any of the fields is private
...
22 |     let _ = datalog::MoreTest(false);
   |                      ^^^^^^^^ private tuple struct constructor
   |
note: the tuple struct constructor `MoreTest` is defined here
  --> $DIR/bad_visibility_of_relation.rs:12:9
   |
12 |         pub struct MoreTest(bool);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^
